modules/images/image-loading-optimization/storage/lock.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Metrics storage lock.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Gets the TTL for the page metric storage lock.
+ *
+ * @return int TTL.
+ */
+function ilo_get_page_metric_storage_lock_ttl() {
+
+	/**
+	 * Filters how long a given IP is locked from submitting another metric-storage REST API request.
+	 *
+	 * Filtering the TTL to zero will disable any metric storage locking. This is useful during development.
+	 *
+	 * @param int $ttl TTL.
+	 */
+	return (int) apply_filters( 'ilo_metrics_storage_lock_ttl', MINUTE_IN_SECONDS );
+}
+
+/**
+ * Gets transient key for locking page metric storage (for the current IP).
+ *
+ * @todo Should the URL be included in the key? Or should a user only be allowed to store one metric?
+ * @return string Transient key.
+ */
+function ilo_get_page_metric_storage_lock_transient_key() {
+	$ip_address = $_SERVER['HTTP_X_FORWARDED_FOR'] ?? $_SERVER['REMOTE_ADDR'];
+	return 'page_metrics_storage_lock_' . wp_hash( $ip_address );
+}
```

Comment:
I don't think it would actually create that many transients because the intention is that metrics would only be gathered when they are needed. So if no metrics are needed, then a frontend request would not create a transient. I believe in the worst case that there would be the total number of transients would be the sample size times the total number of URLs.

Reply:
@westonruter Right, that can be _a lot_ of transients though, for sites with lots of content.

I think as long as we keep these transients with an expiration (so that they disappear over time and are not autoloaded as options), it should be okay.